tsnet/tsnet.go

```diff
@@ -908,6 +908,34 @@ func (s *Server) Listen(network, addr string) (net.Listener, error) {
 	return s.listen(network, addr, listenOnTailnet)
 }
 
+// ListenPacket announces on the Tailscale network.
+//
+// The network must be "udp", "udp4" or "udp6". The addr must be of the form
+// "ip:port" where ip is a valid IPv4 or IPv6 address corresponding to "udp4" or
+// "udp6" respectively. IP must be specified.
+//
+// If s has not been started yet, it will be started.
+func (s *Server) ListenPacket(network, addr string) (net.PacketConn, error) {
+	if err := s.Start(); err != nil {
+		return nil, err
+	}
+	ap, err := resolveListenAddr(network, addr)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
For example, what would you think about something like this?

```suggestion
	}
	// caller did not specify an IP, so use the node's IP based on the specified network
	if !ap.Addr().IsValid() && (network == "udp4" || network == "udp6") {
		st, err := s.Up(context.Background())
		if err != nil {
			return nil, err
		}
		for _, ip := range st.TailscaleIPs {
			if (network == "udp4" && ip.Is4()) || (network == "udp6" && ip.Is6()) {
				ap = netip.AddrPortFrom(ip, ap.Port())
				break
			}
		}
	}
```

Reply:
decided not to do this as it becomes unclear what the expectation is if the IP is changed via the admin console